spec/abci/abci++_app_requirements.md

```diff
@@ -23,15 +23,18 @@ returns via `ResponsePrepareProposal` to CometBFT, also known as the prepared pr
 
 Process *p*'s prepared proposal can differ in two different rounds where *p* is the proposer.
 
-* Requirement 1 [`PrepareProposal`, timeliness]: If *p*'s Application fully executes prepared blocks in
-  `PrepareProposal` and the network is in a synchronous period while processes *p* and *q* are in *r<sub>p</sub>*,
+* Requirement 1 [`PrepareProposal`, timeliness]: If the network is in a synchronous period while processes 
+  *p* and *q* are in *r<sub>p</sub>*,
   then the value of *TimeoutPropose* at *q* must be such that *q*'s propose timer does not time out
   (which would result in *q* prevoting `nil` in *r<sub>p</sub>*).
 
-Full execution of blocks at `PrepareProposal` time stands on CometBFT's critical path. Thus,
+`PrepareProposal` stands on CometBFT's critical path. Thus,
 Requirement 1 ensures the Application or operator will set a value for `TimeoutPropose` such that the time it takes
-to fully execute blocks in `PrepareProposal` does not interfere with CometBFT's propose timer.
-Note that violation of Requirement 1 may just lead to further rounds, but will not compromise liveness.
+to fully execute `PrepareProposal` does not interfere with CometBFT's propose timer.
+Note that violation of Requirement 1 may just lead to further rounds, but will not compromise correctness.
+
+Requirement 1 is particularly important if *p*'s Application fully executes prepared blocks in `PrepareProposal`
+as a form of optimistic execution.
```

Comment:
```suggestion
as a form of immediate execution.
```

Reply:
I believe that optimistic is more accurate here, because it signals that the execution may be wasted, while immediate does not. It is also a term commonly used for similar execution of transactions in DBs. I understand that we have used immediate in some other places so far, though, so would need to review them.

Reply:
I tend to agree with Lasaro, immediate sounds like the execution is final and the state change is persisted.